src/payment/strategies/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -0,0 +1,55 @@
+import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { MissingDataError, MissingDataErrorType } from '../../../common/error/errors';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import { PaymentArgumentInvalidError } from '../../errors';
+import PaymentActionCreator from '../../payment-action-creator';
+import { PaymentRequestOptions } from '../../payment-request-options';
+import PaymentStrategy from '../payment-strategy';
+
+export default class PaypalCommercePaymentStrategy implements PaymentStrategy {
+
+    constructor(
+        private _store: CheckoutStore,
+        private _orderActionCreator: OrderActionCreator,
+        private _paymentActionCreator: PaymentActionCreator
+    ) {}
+
+    initialize(): Promise<InternalCheckoutSelectors> {
+        return Promise.resolve(this._store.getState());
+    }
+
+    execute(payload: OrderRequestBody, options: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = payload;
+        const paymentMethod = this._store.getState().paymentMethods.getPaymentMethod(options.methodId);
+
+        if (!payment) {
+            throw new PaymentArgumentInvalidError(['payment']);
+        }
+
+        if (!paymentMethod || !paymentMethod.initializationData.orderId) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        const paymentData =  {
+            formattedPayload: {
+                vault_payment_instrument: null,
+                device_info: null,
+                paypal_account: {
+                    order_id: paymentMethod.initializationData.orderId,
+                },
+            },
+        };
+
+        return this._store.dispatch(this._orderActionCreator.submitOrder(order, options))
```

Comment:
Prefer `async/await` for new code. 🍹 

PS: I know in some part of our code we use `async/await`, and in others, we use the regular `then` promise chain. The reason is simply because the former is newer code, whereas the latter is older code.